# Assignment 3

### Instructions
1. Import the following modules:
    - `pandas`
    - `stats` from `scipy`
2. Create a dataframe from the data contained in *income_data.csv* using `pandas`

In [1]:
import pandas as pd
from scipy import stats

df = pd.read_csv("income_data.csv")
print(f"Number of observations: {len(df)}")
print(df.head().to_string(index=False))

Number of observations: 45
       job type  income  education  prestige
accountant prof      62         86        82
     pilot prof      72         76        83
 architect prof      75         92        90
    author prof      55         90        76
   chemist prof      64         86        90


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *education*

In [2]:
r_income_education, p_income_education = stats.pearsonr(df["income"], df["education"])
print("Income and education")
print(f"Pearson's r: {r_income_education:.6f}")
print(f"p-value (two-sided): {p_income_education:.6e}")

Income and education
Pearson's r: 0.724512
p-value (two-sided): 1.839946e-08


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *prestige*

In [3]:
r_income_prestige, p_income_prestige = stats.pearsonr(df["income"], df["prestige"])
print("Income and prestige")
print(f"Pearson's r: {r_income_prestige:.6f}")
print(f"p-value (two-sided): {p_income_prestige:.6e}")

Income and prestige
Pearson's r: 0.837801
p-value (two-sided): 7.144356e-13


4. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *education* and *prestige*

In [4]:
r_education_prestige, p_education_prestige = stats.pearsonr(df["education"], df["prestige"])
print("Education and prestige")
print(f"Pearson's r: {r_education_prestige:.6f}")
print(f"p-value (two-sided): {p_education_prestige:.6e}")

Education and prestige
Pearson's r: 0.851916
p-value (two-sided): 1.170879e-13


5. In the empty `Markdown` cell below, explain the following for **each of the calculated correlation coefficients**
    - The **strength** and **direction** of the correlation
    - The **statistical significance** of the test (**p-value**); emphasize if we can trust the correlation test or not, and why

All three correlations are strong and positive:
Income-education: r = 0.725, p = 1.84e-08.
Income-prestige: r = 0.838, p = 7.14e-13.
Education-prestige: r = 0.852, p = 1.17e-13.

All are significant (p < 0.05). The tests are trustworthy if independence and bivariate normality hold and influential outliers are absent; small p-values do not verify these assumptions or prove causation.

In the empty `Markdown` cell below, answer the following question:

6. Pearson's correlation coefficient OR covariance can both be used to measure the direction of the relationship between two variables (e.g., education and income). Explain why Pearson's correlation coefficient may be better than covariance.

Pearson's r is unitless and ranges from -1 to +1, making direction and linear strength easy to interpret and compare. Covariance depends on units and has no fixed range.

7. Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment3.ipynb** 
    - Replace '**Last**' and '**First**' with your name